src/nifgen/examples/nifgen_script.py

```diff
@@ -2,52 +2,43 @@
 
 import argparse
 import nifgen
-import numpy as np
-from scipy import signal
+
+import math
+import random
 import sys
 import time
 
+# waveform size should be a multiple of the quantum, which is 4 or 8, for some devices
+# some older devices also have a 256 sample minimum waveform size
 number_of_points = 256
 
 
 def calculate_sinewave():
-    time = np.linspace(start=0, stop=10, num=number_of_points)    # np.linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None)
-    amplitude = np.sin(time)
-    sinewave = amplitude.tolist()                               # List of Float
-    return sinewave
+    # waveforms finish just short of 360 degrees, so that we don't repeat the first point
+    # if we repeat the waveform
+    return [math.sin(math.pi * 2 * x / (number_of_points)) for x in range(number_of_points)]
 
 
 def calculate_rampup():
-    ramp = np.linspace(start=0, stop=0.5, num=number_of_points)   # np.linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None)
-    ramp_up = ramp.tolist()                                     # List of Float
-    return ramp_up
+    return [x / (number_of_points) for x in range(number_of_points)]
 
 
 def calculate_rampdown():
-    ramp = np.linspace(start=0, stop=0.5, num=number_of_points)   # np.linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None)
-    ramp_down = ramp.tolist()                                   # List of Float
-    ramp_down.reverse()                                         # Reverse list to get a ramp down
-    return ramp_down
+    return [-1.0 * x for x in calculate_rampup()]
 
 
 def calculate_square():
-    time = np.linspace(start=0, stop=10, num=number_of_points)    # np.linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None)
-    square_build = signal.square(t=time, duty=0.5)              # signal.square(t, duty=0.5)
-    square = square_build.tolist()                              # List of Float
-    return square
+    return [1.0 if x < (number_of_points / 2) else -1.0 for x in range(number_of_points)]
 
 
-def calculate_triangle():
-    time = np.linspace(start=0, stop=1, num=number_of_points)     # np.linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None)
-    triangle_build = signal.sawtooth(t=time)                    # signal.sawtooth(t, width=1)
-    triangle = triangle_build.tolist()                          # List of Float
-    return triangle
+def calculate_sawtooth():
```

Comment:
It's weird / surprising that this function returns something other than `number_of_points`.
Surprising is bad.

Reply:
Replaced the functions with CONSTANTS.